Feature/resource management cdn (feature mittel)

notefox opened the pull request:
## User Resource Creation und Management + Azure CDN

### Feature beinhaltet:
 - erstellen von User Resources gebunden mit User
 - gruppieren von User Resources mit angebundenen custom Tags, welcher der User selbst erstellen kann (suchen und sortieren kommt mit feature/search)
 - hochladen und runterladen von den angebundenen Files im zusammenspiel mit Microsoft Auzre und Active Storage
 - CRUD implementation von den vorherig benannten Resources
 - erweiterung und anpassung vom Dashboard an die neuen Resources
 - visuel leicht zu verstehendes Desing mit klarem Stil
 - angenehme User Experience, getestet an einer weitreichenden Nutzer gruppe
 - angepasst an anzustehenden Pull der User Authentication und 
 - gegebene basis für die bald angehende Search Funktion des nächsten Pull Requests feature/search

[ Gruppe 1, s0569194 ]

Commit:
split up dashboard index.html.erb code into extra html files:
 - _categories_listing.html.erb
 - _recent_files_list.html.erb
 - _selected_file_listing.html.erb

Commit:
split up dashboard index.html.erb code into extra html files:
 - _categories_listing.html.erb
 - _recent_files_list.html.erb
 - _selected_file_listing.html.erb

Commit:
rebase origin/main

Commit:
formatting changes

Commit:
routes.rb re-managing

Commit:
routes.rb re-managing

Commit:
Rewrite FileResource to Resource

Commit:
Rewrite FileResource to Resource

Commit:
information dump page for all Resources

Commit:
information dump page for all Resources

Commit:
add forgotten resource tests and helper files

Commit:
add forgotten resource tests and helper files

Commit:
add forgotten resource tests and helper files

Ktechen committed:
Add Debug overview

Ktechen committed:
Add Debug overview

Commit:
remove unused file_resource_relation
add azure-storage-blob gem

Commit:
remove unused file_resource_relation
add azure-storage-blob gem

Commit:
recreate new resources

Commit:
recreate new resources

Commit:
add session helper
add resource helper

Commit:
add session helper
add resource helper

Commit:
add session helper
add resource helper

Commit:
error-controller

Commit:
error-controller

Commit:
add new resource model with scaffold-option

Commit:
add new resource model with scaffold-option

Commit:
rebase merge conflict resolution

Commit:
add active storage tables for active storage over azure

Commit:
add active storage tables for active storage over azure

Commit:
resolved merge conflicts

Commit:
rubocop changes

Commit:
rubocop changes

Commit:
resolved merge conflicts

Commit:
fixed double rendering

Commit:
fixed double rendering

Commit:
remove faulty session_helper.rb

Commit:
remove faulty session_helper.rb

Commit:
remove faulty session_helper.rb

Commit:
fix multiple loadings of bootstrap
 - add *=require _bootstrap to application.css

Commit:
fix multiple loadings of bootstrap
 - add *=require _bootstrap to application.css

Commit:
rebase merge conflict resolutions

notefox committed:
file_resource -> user_resource

notefox committed:
file_resource -> user_resource

notefox committed:
add tags

notefox committed:
add tags

notefox committed:
add missing new user_resource view

notefox committed:
add missing new user_resource view

notefox committed:
add missing migration files for user_resource and tag

notefox committed:
add missing migration files for user_resource and tag

notefox committed:
remove old un-needed files

notefox committed:
remove old un-needed files

notefox committed:
rubocop

notefox committed:
rubocop

notefox committed:
git rebase merge conflict resolution

notefox committed:
change outdated user_resource migration file

notefox committed:
change outdated user_resource migration file

notefox committed:
update migration files

notefox committed:
update migration files

notefox committed:
rewrite faulty migration file

notefox committed:
rewrite faulty migration file

notefox committed:
rewrite faulty migration file

notefox committed:
update removed files on git

notefox committed:
update removed files on git

notefox committed:
update removed files on git

notefox committed:
reformatting commit

notefox committed:
reformatting commit

notefox committed:
update application.scss

notefox committed:
update application.scss

notefox committed:
add user_resource.rb

notefox committed:
add user_resource.rb

notefox committed:
add user_resource.rb

notefox committed:
remove old relics

notefox committed:
remove old relics

notefox committed:
add tests

notefox committed:
add tests

notefox committed:
add tests

notefox committed:
update docker script (just a directory change due to an update)

notefox committed:
update docker script (just a directory change due to an update)

notefox committed:
add seed

notefox committed:
add seed

notefox committed:
add forgotten dashboard files

notefox committed:
add forgotten dashboard files

notefox committed:
add forgotten dashboard files

notefox committed:
update background image

notefox committed:
update background image

notefox committed:
add tag

notefox committed:
add tag

notefox committed:
add debugging functionality configuration

notefox committed:
add debugging functionality configuration

notefox committed:
add new routes

notefox committed:
add new routes

notefox committed:
add new routes

notefox committed:
update gem file

notefox committed:
update gem file

notefox committed:
update gem file

notefox committed:
add test fixtures

notefox committed:
add test fixtures

notefox committed:
add tag user resource relation table migration file

notefox committed:
add tag user resource relation table migration file

notefox committed:
add possible placeholder image

notefox committed:
add possible placeholder image

notefox committed:
add extra files for rebase

notefox committed:
Merge remote-tracking branch 'origin/feature/ressource-management_CDN' into feature/ressource-management_CDN

# Conflicts:
#	Gemfile.lock
#	config/routes.rb
#	db/schema.rb

notefox committed:
rebase complete

notefox committed:
rebase complete

notefox committed:
codacy updates

notefox committed:
codacy updates

notefox committed:
readd user

notefox committed:
readd user

notefox committed:
git rebase merge conflict resolution

notefox committed:
add form field required: true

Ktechen commented:
Woher kommen die ganzen empty files?

NoteFox commented:
> Woher kommen die ganzen empty files?

rebase

notefox committed:
rebase conflict resolutions

notefox committed:
rebase conflict resolutions

notefox committed:
git rebase merge conflict resolution

notefox committed:
git rebase merge conflict resolution

notefox committed:
Merge remote-tracking branch 'origin/feature/ressource-management_CDN' into feature/ressource-management_CDN

# Conflicts:
#	Gemfile.lock

Ktechen commented on app/controllers/user_resources_controller.rb:
```diff
@@ -0,0 +1,103 @@
+  # POST /user_resources or /user_resources.json
+  def create
+    @user = session[:user_id]
+    redirect_to registrations_index_path if @user.nil?
```
Du benutzt den Code 3 mal, dann mach doch eine private Methode.

notefox replied:
mach ich

viktorianer commented:
Auf welchen Brach hast du hier aufgebaut? Das schaut für mich so zu sein, dass du hier sehr viele Files aus einem anderen PR hast?

NoteFox commented:
> Auf welchen Brach hast du hier aufgebaut? Das schaut für mich so zu sein, dass du hier sehr viele Files aus einem anderen PR hast?

Der Branch wurde auf der damaligen Basis des Main gebaut, bevor der header / footer existent war, 
danach gab es 2 rebases vom main auf den hier gegebenen Branch, damit dieser Branch auf dem gleichen  Stand mit dem aktuellen main ist, um sicher zu gehen, dass es dann beim merge keine probleme aufkommen

viktorianer commented:
> > Auf welchen Brach hast du hier aufgebaut? Das schaut für mich so zu sein, dass du hier sehr viele Files aus einem anderen PR hast?
> 
> Der Branch wurde auf der damaligen Basis des Main gebaut, bevor der header / footer existent war,
> danach gab es 2 rebases vom main auf den hier gegebenen Branch, damit dieser Branch auf dem gleichen Stand mit dem aktuellen main ist, um sicher zu gehen, dass es dann beim merge keine probleme aufkommen

Aber du willst doch nicht sagen, dass hier `resource management cdn` 116 Commits und 174 Filechanges brauchte? xD

NoteFox commented:
> > > Auf welchen Brach hast du hier aufgebaut? Das schaut für mich so zu sein, dass du hier sehr viele Files aus einem anderen PR hast?
> > 
> > 
> > Der Branch wurde auf der damaligen Basis des Main gebaut, bevor der header / footer existent war,
> > danach gab es 2 rebases vom main auf den hier gegebenen Branch, damit dieser Branch auf dem gleichen Stand mit dem aktuellen main ist, um sicher zu gehen, dass es dann beim merge keine probleme aufkommen
> 
> Aber du willst doch nicht sagen, dass hier `resource management cdn` 116 Commits und 174 Filechanges brauchte? xD

nein, brauchte es bestimmt nicht ^^"
ich gehe davon aus das meiste ist rebase und rubocop

viktorianer commented:
> nein, brauchte es bestimmt nicht ^^"
> ich gehe davon aus das meiste ist rebase und rubocop

Dann mache `rubocop` ein mal auf dem Master und merge es hier rein. Ich will nicht 170 Datei durchgehen, die nichts mit diesem Feature zu tun haben. Wie in Vorlesung gesagt, im PRs sollten nur Feature abgegeben werden. Also nur das, was zum Feature gehört.

viktorianer commented:
Alle Konfigurationen und Upgrades die hier wohl auch gemacht wurden, bitte auf dem Master machen oder von hier mit `git cherry-pick` holen und danach hier reinmergen. 

Sobald dieses PR auf eine angemessene Größe runter ist, schaue ich es an.

notefox committed:
try to shorten pull request

NoteFox commented:
> Alle Konfigurationen und Upgrades die hier wohl auch gemacht wurden, bitte auf dem Master machen oder von hier mit `git cherry-pick` holen und danach hier reinmergen.
> 
> Sobald dieses PR auf eine angemessene Größe runter ist, schaue ich es an.

wie siehts jetzt aus?

notefox committed:
try to shorten pull request

Ktechen commented:
Von 174 files auf 76 files nice da haste Aufgeräumt

NoteFox commented:
> Von 174 files auf 76 files nice da haste Aufgeräumt

und das mit nur einem python script das ich in 15 minuten zusammengeworfen habe

viktorianer reviewed (approved):
Also, wichtigstes zu erst: ist das nun alles von dir?

Wenn ja, bist du durch. Brauchst nichts weiter zu machen. 

Inhaltlich ist alles OK.

Aber, was du für Zukunft merken solltest:
Konzentriere dich und bleib bei einer Sache!

Die Arbeit hier beinhaltet völliges durcheinander an Konfiguration, Änderungen die nichts mit einem Feature alleine zu tun haben. Du springst von einem File zu anderen, ohne Ziel, so kommt mir das vor.

Das führt automatisch zu Fehler, die du gar nicht machen musst.

Dafür muss ich Punkte abziehen.

NoteFox commented:
> Also, wichtigstes zu erst: ist das nun alles von dir?
> 
> Wenn ja, bist du durch. Brauchst nichts weiter zu machen.
> 
> Inhaltlich ist alles OK.
> 

ja das ist alles meines
okay, freue mich erstmal das ich durch bin

> Die Arbeit hier beinhaltet völliges durcheinander an Konfiguration, Änderungen die nichts mit einem Feature alleine zu tun haben. Du springst von einem File zu anderen, ohne Ziel, so kommt mir das vor.
> 
> Das führt automatisch zu Fehler, die du gar nicht machen musst.

Beziehen sie sich hier auf meine Commit History, oder nur auf die veränderten Dateien im Allgemeinen?

viktorianer commented:
> > Von 174 files auf 76 files nice da haste Aufgeräumt
> 
> und das mit nur einem python script das ich in 15 minuten zusammengeworfen habe

Nun, mit sauberer Arbeit hättest du keinen Skript gebraucht :P

viktorianer commented:
> Beziehen sie sich hier auf meine Commit History, oder nur auf die veränderten Dateien im Allgemeinen?

Auf beides.

viktorianer commented:
Ahja, nur damit du dich danach nicht wunders: Punkte für Tests habe ich nicht vergeben, da bis jetzt keine Tests dabei waren. Für keinen der Features.

NoteFox commented:
> Ahja, nur damit du dich danach nicht wunders: Punkte für Tests habe ich nicht vergeben, da bis jetzt keine Tests dabei waren. Für keinen der Features.

naja, tests gibts bei mir schon, nur halt noch nicht so viele
(außerdem wurden galube für [diesen Pull Request hier](https://github.com/Ktechen/Dripbox/pull/26) schon tests geschrieben)

viktorianer commented:
> naja, tests gibts bei mir schon, nur halt noch nicht so viele
> (außerdem wurden glaube für [diesen Pull Request hier](https://github.com/Ktechen/Dripbox/pull/26) schon tests geschrieben)

Das sind nicht deine Tests. In deinen PRs gibt es keine Tests.

notefox committed:
minor fixes and changes for testing:
- remove old files
- _selected_file_listing.html.erb produced errors due to a name change
- webpacker.yml complie : true
- seed remake
- removed some old, not usable code
- add some tests

notefox committed:
tests added

Commit:
the "nothing" commit

Commit:
pull request merge conflict resolve

Commit:
Merge branch 'main' of https://github.com/Ktechen/Dripbox into feature/ressource-management_CDN

 Conflicts:
	Gemfile.lock
	db/schema.rb
